care/emr/api/viewsets/diagnostic_report.py

```
@@ -198,4 +199,74 @@ def upsert_observations(self, request, *args, **kwargs):
             model_instance.diagnostic_report = diagnostic_report
             model_instance.subject_type = SubjectType.encounter.value
             model_instance.save()
+
+            # Compute interpretation if observation_definition is linked
+            if model_instance.observation_definition:
+                patient = model_instance.patient
+                age = None
+                if patient.date_of_birth and model_instance.effective_datetime:
+                    if isinstance(model_instance.effective_datetime, str):
+                        from datetime import datetime
```

Ruff (0.12.2)

`import` should be at the top-level of a file

(PLC0415)